### DISCUSSION TITLE
Easy way to remove Prose styles from build

### DISCUSSION THREAD

**Discussion by darcemac** [gregsullivan/_tw#62] [General]
Hi there,

Fantastic theme by the way, made building my last site a pleasure!

I was wondering whats the best way to remove all Tailwind Prose styles as its playing havoc visually with my ACF blocks in the Editor.

Classes like the one below are just not needed.

`.prose :where(h2):not(:where([class~="not-prose"] *))`

**Comment by gregsullivan** [comment]
Hi Darcy,

Very happy to hear _tw worked well for you on the last site you built!

The prose styles are coming from [Tailwind Typography](https://tailwindcss.com/docs/typography-plugin). If your concern is just removing the CSS, you can remove these lines from `./tailwind/tailwind.config.js`:

```javascript
presets: [
	// Manage Tailwind Typography's configuration in a separate file.
	require('./tailwind-typography.config.js'),
],
```
```javascript
// Add Tailwind Typography.
require('@tailwindcss/typography'),
```

You can also delete the `./tailwind/tailwind-typography.config.js` file entirely.

Then, if you don't want the extra classes being added to your frontend HTML, you can set the value of `EVKEEZA_TYPOGRAPHY_CLASSES` to an empty string in `./theme/functions.php`.

Finally, you can also comment out this line in `./theme/functions.php`:

```php
add_action( 'enqueue_block_assets', 'evkeeza_enqueue_typography_script' );
```

(You could also delete the entire `evkeeza_enqueue_typography_script()` function and the corresponding `add_action`—up to you!)

**Reply by darcemac** [reply]
Awesome thanks your response, much appreciated! I'm new to TW, this was literally the first site developed using it...

So now I'm getting more type styles in page editor that are coming from somewhere. Is this because I haven't specified enough classes in the frontend to cancel them out? eg. mt-0, fw-normal, color-black

```
html :where(.editor-styles-wrapper) h1, html :where(.editor-styles-wrapper) h2, html :where(.editor-styles-wrapper) h3, html :where(.editor-styles-wrapper) h4, html :where(.editor-styles-wrapper) h5, html :where(.editor-styles-wrapper) h6 {
  color: revert;
  font-size: revert;
  font-weight: revert;
  line-height: revert;
  margin: revert;
    margin-bottom: revert;
}
```

Im really not sure why all header elements need margins top and bottom.

**Reply by gregsullivan** [reply]
Sorry for the delay in responding! I'm working limited hours this week.

Can you clarify where you're seeing that CSS? Nothing in the output from _tw should include `.editor-styles-wrapper` (unless, maybe, if you're looking at styles modified by the block editor).

**Reply by darcemac** [reply]
Hey no worries at all, thanks for your response!

So my query is because Im trying to get what is in the Page editor as close to possible as what it looks like on the frontend. I realise its prob not 100% possible but thought I would try.

Im using ACF blocks fwiw and most of my styles are coming from inline classes but some are coming from the custom.css file.

The example Ive attached is of a link where the styles arent being transferred across to the page editor. The link should have a orange border and there shouldn't be an underline.

![Screen Shot 2023-09-01 at 5 41 41 pm](https://github.com/gregsullivan/_tw/assets/1846302/6dc7d810-2196-43c9-b206-8e2855d0c50f)

![Screen Shot 2023-09-01 at 5 41 48 pm](https://github.com/gregsullivan/_tw/assets/1846302/3dade759-5a5e-4896-b917-bcc4149e6acb)

**Reply by gregsullivan** [reply]
Can you confirm whether the orange border is being applied and then overridden by something, or whether it's never being applied at all?

I've experienced something similar to the underline issue: There are some editor styles that you'll need to explicitly override. Since Tailwind removes the underlines in its Preflight, but we don't use Preflight in the editor, you'll need to add `no-underline` if you want to remove it. There aren't a huge number of things like that, but that's definitely one I've seen before.

**Reply by gregsullivan** [reply]
It could also be an example of this for the border:

https://github.com/gregsullivan/_tw/blob/55a031600777b3bdcb1116b750ee7ac74c2eacf6/tailwind/tailwind-typography.config.js#L24-L32

**Reply by darcemac** [reply]
Hi Greg,

I believe the link is getting the underline and perhaps the border width 0 from the below inline styles

```
.editor-styles-wrapper a:where(:not(.wp-element-button)) {
text-decoration: underline;
}

.editor-styles-wrapper .wp-element-button,
.editor-styles-wrapper .wp-block-button__link {
background-color: #32373c;
border-width: 0px;
color: #fff;
font-family: inherit;
font-size: inherit;
line-height: inherit;
padding: calc(0.667em + 2px) calc(1.333em + 2px);
text-decoration: none;
}
```
When I change it to a button element it get the borders back and the underline is gone.

**Reply by darcemac** [reply]
Thanks for the typography.config.js link. Perhaps I need to create a custom link class for this?

Im learning a lot about TW and have to say the development process has been a lot more fun using your theme, so thanks very much!

**Reply by gregsullivan** [reply]
I think you're right about that being where the underline is coming from! You might want to try adding `no-underline` to your classes for the link, as Tailwind won't be removing the underlines without Preflight.

You could check in dev tools whether the border is zero-width, or whether there's no border style. If it's the latter, adding `border-solid` might fix the issue.

I'll note that the `typography.config.js` file isn't meant to be used to style things like ACF blocks—it's specifically for tweaking the output from Tailwind Typography when you need to go beyond what the [element modifiers](https://tailwindcss.com/docs/typography-plugin#element-modifiers) allow.

I'm very happy to hear you've been enjoying the development process—you're very welcome!
